azure/src/wizard/ResourceGroupVerifyStep.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ResourceManagementClient } from '@azure/arm-resources';
+import { ActivityChildItem, ActivityChildType, activityFailContext, activityFailIcon, AzureWizardExecuteStepWithActivityOutput, createContextValue, ExecuteActivityOutput, nonNullProp, parseError } from '@microsoft/vscode-azext-utils';
+import { v4 as uuidv4 } from "uuid";
+import { l10n, Progress, TreeItemCollapsibleState } from 'vscode';
+import * as types from '../../index';
+import { createResourcesClient } from '../clients';
+import { ext } from '../extensionVariables';
+
+// See for more background: https://github.com/microsoft/vscode-azuretools/pull/1992#issue-3034841865
+export class ResourceGroupVerifyStep<T extends types.IResourceGroupWizardContext> extends AzureWizardExecuteStepWithActivityOutput<T> {
+    public priority: number = 95;
+    public stepName: string = 'resourceGroupVerifyStep';
+
+    protected getOutputLogFail = (context: T) => l10n.t('Failed to verify whether resource group with name "{0}" already exists.', nonNullProp(context, 'newResourceGroupName'));
+    protected getOutputLogSuccess(context: T) {
+        return context.resourceGroup?.name ?
+            l10n.t('Verified resource group with name "{0}" already exists.  Using existing resource group.', context.resourceGroup.name) :
+            l10n.t('Verified resource group with name "{0}" is available for creation.', nonNullProp(context, 'newResourceGroupName'));
+    }
+    protected getTreeItemLabel(context: T) {
+        return context.resourceGroup?.name ?
+            l10n.t('Verify resource group "{0}" exists', context.resourceGroup.name) :
+            l10n.t('Verify resource group "{0}" available', nonNullProp(context, 'newResourceGroupName'));
```

Comment:
```suggestion
            l10n.t('Verify resource group "{0}" is available', nonNullProp(context, 'newResourceGroupName'));
```

Reply:
Actually, I wouldn't recommend using nonNullProp here. Just in case it's empty for some weird reason, rather have the tree label just be undefined or "" instead of throwing an error.